### PR TITLE
#161754577 Add get comment edit history feature

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
- Add GET comment edit history feature

#### Description of Task to be completed?
- Create the route for getting a comment and its edit history
- Add tests for route
- write documentation
- Update migration for Comments and CommentHistories table

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on postman
- run `npm run start-dev`
- *For Comment,* On postman run GET `localhost:5000/api/articles/:slug/comments/:id` (set JWT token as`x-access-token` in header).

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#161754577

#### Screenshots (if appropriate)

Good Response 1:
screen shot 2018-11-08 at 2 38 28 pm

Good Response 2 (No edit history):
screen shot 2018-11-08 at 3 00 05 pm

Bad Request Response (when a comment is not found):
screen shot 2018-11-07 at 1 45 58 pm

### CODE REVIEW

**Comment by ucheg6** [comment]
Good work Theo. This looks clean

File: server/repository/commentRepository.js [hunk 1]
```diff
@@ -63,6 +63,35 @@ class CommentRepository {
...
+   * otherwise it throws an error
+   */
+  static async getCommentWithHistory(id) {
+    const commentRecord = await Comment.findOne({
+      where: {
+        id,
+      },
+      include: [{
+        model: CommentHistory,
+        as: 'editHistory',
```
Comment (PriscillaSam) [inline server/repository/commentRepository.js:80]: @TheoOkafor can this just be `include: ['editHistory']`? Did you try it?

File: server/models/commenthistory.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ export default (sequelize, DataTypes) => {
   CommentHistory.associate = (models) => {
     CommentHistory.belongsTo(models.Comment, {
       foreignKey: 'commentId',
+      useJunctionTable: false,
```
Comment (PriscillaSam) [inline server/models/commenthistory.js:15]: @TheoOkafor why are we using `useJunctionTable: false` here? I thought this was for the one to many relationship between comment and comment?

**Comment by PriscillaSam** [comment]
@TheoOkafor well done. This looks great. Well done. 👍

File: server/repository/commentRepository.js (hunk 1 above)
Comment (TheoOkafor) [reply server/repository/commentRepository.js:80]: I did. I changed it later. I feel that the code will be easier to understand this way.

File: server/models/commenthistory.js (hunk 2 above)
Comment (TheoOkafor) [reply server/models/commenthistory.js:15]: @PriscillaSam It is set to `false`. I am not sure about the Sequelize default value for the `useJunctionTable`. I had to state the required value explicitly.

**Comment by TheoOkafor** [comment]
Thank you @ucheg6 and @PriscillaSam

File: server/controllers/comment.js [hunk 3]
```diff
@@ -81,6 +81,38 @@ class Comment {
...
     return goodHttpResponse(response, 200, 'Comment updated', updatedComment);
   }
+
+  /** Create a new comment
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} Comment Object
+   */
+  static async getCommentWithHistory(request, response) {
+    if (!request.isAuthorized) {
```
Comment (seyi-adeleke) [inline server/controllers/comment.js:98]: is this endpoint only accessible to the whoever left the origin comment?

File: server/controllers/comment.js [hunk 4]
```diff
@@ -81,6 +81,38 @@ class Comment {
...
+      return badHttpResponse(
+        response,
+        401,
+        'You are not permitted to complete this action',
+      );
+    }
+
+    const { slug, id } = request.params;
+    const article = await articleRepo.getArticleBySlug(slug);
```
Comment (seyi-adeleke) [inline server/controllers/comment.js:107]: When an article is deleted is there a cascade that deletes all the comments that belong to the article?

File: server/controllers/comment.js (hunk 3 above)
Comment (TheoOkafor) [reply server/controllers/comment.js:98]: Yes

File: server/controllers/comment.js (hunk 4 above)
Comment (TheoOkafor) [reply server/controllers/comment.js:107]: No. I will update that now.

Comment (TheoOkafor) [reply server/controllers/comment.js:107]: Fixed

File: server/migrations/20181113025401-update-articleid-column.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export default {
+  up: queryInterface => queryInterface.sequelize.query('alter table "Comments" drop constraint "Comments_articleId_fkey"')
+    .then(() => queryInterface.sequelize.query(
```
Comment (seyi-adeleke) [inline server/migrations/20181113025401-update-articleid-column.js:3]: whyyyyy?

Comment (seyi-adeleke) [reply server/migrations/20181113025401-update-articleid-column.js:3]: Whats the advantage of writing raw queries here?

Comment (TheoOkafor) [reply server/migrations/20181113025401-update-articleid-column.js:3]: The Sequelize `updateColumn` method did not work. For some reason, it adds the updated foreign key to the column without removing the old foreign key attribute.
Raw queries solved this problem.

**Pull request merged**